file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs

```diff
@@ -300,12 +352,45 @@ private async Task<string> HandleCanceledOperationsAsync(IBatchHandle batchHandl
             return ExecutionResult;
         }
 
+        private async Task RollBackBatch(IBatchHandle batchHandle, IFileShareApiAdminClient fileShareClient)
+        {
+            logger.LogInformation("File Share Service RollBackBatchAsync started for batch ID:{BatchId}.", batchHandle.BatchId);
+            var rollBackBatchResult = await fileShareClient.RollBackBatchAsync(batchHandle, CancellationToken.None);
+            if (rollBackBatchResult.IsSuccess)
+            {
+                ExecutionResult = $"File Share Service rollback completed for batch ID: {batchHandle?.BatchId}";
+            }
+            else
+            {
+                ExecutionResult = (rollBackBatchResult.Errors != null && rollBackBatchResult.Errors.Any()) ?
+                    string.Join(Environment.NewLine, rollBackBatchResult.Errors.Select(e => e.Description)) :
+                    $"File Share Service rollback failed for batch ID:{batchHandle?.BatchId} with status: {rollBackBatchResult.StatusCode}.";
+
+                logger.LogError("File Share Service rollback failed for displayName:{DisplayName} and batch ID:{BatchId} with error:{responseMessage}.",
+                    DisplayName, batchHandle?.BatchId, ExecutionResult);
+            }
+            logger.LogInformation("File Share Service RollBackBatchAsync completed for batch ID:{BatchId}.", batchHandle.BatchId);
+        }
+
         private async Task SetBatchExpiry(IBatchHandle batchHandle, IFileShareApiAdminClient fileShareClient)
         {
             logger.LogInformation("File Share Service SetExpiryDateAsync started for batch ID:{BatchId}.", batchHandle.BatchId);
-            string expiryDateString = ConvertToRFC3339Format(DateTime.UtcNow.AddDays(-7));
-            await fileShareClient.SetExpiryDateAsync(batchHandle.BatchId, new BatchExpiryModel { ExpiryDate = expiryDateString }, CancellationToken.None);
-            logger.LogInformation("File Share Service SetExpiryDateAsync completed for batch ID:{BatchId}.", batchHandle.BatchId);
+            DateTime? expiryDateString = DateTime.UtcNow.AddDays(-7);
```

Comment:
Please use `currentDateTimeProvider` instead of DateTime.UtcNow -
For test scenarios we need to inject different dates and that is the purpose of `currentDateTimeProvider`